### DISCUSSION TITLE
Abyss creatures respawn immediately after despawn a corpse of previous killed creature

### DISCUSSION THREAD

**Discussion by Xardion495** [FakeFishGames/Barotrauma#13992] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Abyss creatures respawn immediately after despawn a corpse of previous killed creature on abyssal preset. I'm not sure is this a bug because we playing on abyssal preset, so that's possible meant to be.
But if not players can't farm mineral allods calmly on abyssal preset

### Reproduction steps

1. Start multiplayer campaign
2. Achieve progress to 64% difficulty (The Great Sea)
3. Kill endworm
4. Stay near the endworm corpse
5. Face the new endworm after a corpse of previous one disappeared

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by itchyOwl** [comment]
The abyss monster event should reset after 300 seconds (5 minutes) in that random event set. I don't think the abyssal preset changes this. Despawning of the corpse shouldn't affect it in any way. Are you sure it happens every time, or could it just be a coincidence?

**Comment by Xardion495** [comment]
Well.. When you mentioned the event reset value I checked the stream record. Apparently there is no bug cause the respawn time is about 6 minutes. My mistake

Here is record from twitch: https://www.twitch.tv/videos/2149984397
41:40 - Kill the first endworm
47:36 - Game spawn the second one
48:22 - Kill the second one (by admin command)
56:18 - Game spawn the third one

**Comment by TheDarkBox** [comment]
Hey, can confirm that this happens in a completely unmodded campaign.
We were fighting a Charybdis in a part of "The Great Sea" with a 3 person crew.

One of us died and was able to see the new Charybdis spawn in as soon as the previous one's dead body despawned.
This happened twice (so there were a total of 3 Charybdises.. Charybdi (?)) we had to fight, one right after another.

I can send a ~40 minute video clip if needed.
Was not a fun encounter..

**Reply by SomeRandomNoobKekeke** [reply]
> I can send a ~40 minute video clip if needed.

very needed

i'm sure devs won't see this coz it's closed + it will be a lot deeper when devs came back from hibernation in august, so you sould investigate code yourself, create new ticket in august

**Reply by Regalis11** [reply]
We do see these. :)

**Reply by SomeRandomNoobKekeke** [reply]
@Regalis11  [Have you seen those? ](https://github.com/FakeFishGames/Barotrauma/discussions/13787#discussioncomment-9580045)

**Comment by A2ur3** [comment]
it's intended in great sea abyssal monster will respawn after 5 minutes. you are not allowed to stay in the abyss and mine all minerals

**Comment by TheDarkBox** [comment]
Well, perhaps I misunderstood the response.
In any case, what if we did not intend to mine the minerals and accidentally fell in due to unrelated creatures damaging the ship?

Even if you defeat the monster, it essentially softlocks you down there if you can't make it out within 5 minutes (depending on the damage to the sub and the (remaining) crew size, that might be difficult or even impossible).

If this is all to prevent people from mining minerals, I think it would make more sense to have one monster spawn initially as is, and only have it start the respawn timer if it detects a live person outside the submarine after the first monster dies.. or if it detects somebody mining minerals if you want to be fancy.

I understand that it's supposed to be difficult, but the respawn timer seems downright unfair.

**Reply by Regalis11** [reply]
The point of the respawn isn't to prevent mining specifically, it's just to make sure that the abyss is always an extremely dangerous place to be in, even if you've killed one abyss monster. I would argue that the risk of getting softlocked is actually pretty low: even if another monster spawns after the first one, it's far from guaranteed that it'll spot you or attack you as long as you don't make unnecessary noise. There are also plenty of other strategies for mining in the abyss, some of the crew could for example go and mine inside an abyss island while the rest keep the abyss monster busy.

**Reply by TheDarkBox** [reply]
Well, in our playthrough, it felt like the creature respawned right next to the sub on its second respawn.
